src/test/resources/org/rust/ide/intentions/fixtures/unwrap_lambda_expr/unwrap_braces.rs

```diff
@@ -0,0 +1,3 @@
+fn main () {
+    |x| { x *<caret> x }
```

Comment:
Hmm what about (pseudocode):
```kotlin
// find first parent which is either block or lambda
// if we get lambda, than return false as we are inline "unblocked" (hehe) lambda
val block = PsiTreeUtil.findFirstParent(element, blockOrLambda) as? RustBlockExprElement ?: return false

// check whether our block's parent is lambda to avoid false positives
// like nested blocks in lambda (we are not interested in this case as we can
// be sure that cursor isn't at brace; well, I don't know what about `|x| {<caret>{ x }}`,
// but it's possibly rare edge case), or just some random block in code
if (block.parent !is RustLambdaExprElement) return false

// ... in companion object
private val blockOrLambda = Condition<PsiElement> { element ->
    element is RustBlockExprElement || element is RustLambdaExprElement
}
```

I used this trick with `findFirstParent` in [RustContextType](https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/ide/template/RustContextType.kt) and nobody complained yet.